### DISCUSSION TITLE
LDAP Auth question

### DISCUSSION THREAD

**Discussion by rja1** [provectus/kafka-ui#2612] [Q&A] [answered]
Hey Everyone,

First off, I love this tool!  I've been looking for something like this for a long time!  

Question regarding LDAP auth.  I only see 3 parameters to configure it.  IE
      AUTH_TYPE: "LDAP"
      SPRING_LDAP_URLS: "ldap://ldap:10389"
      SPRING_LDAP_DN_PATTERN: "cn={0},ou=people,dc=planetexpress,dc=com"

Typically, LDAP implementations have a username/password to authenticate against the ldap server, authenticate/bind the user and finally a couple parameters to validate if a user is a member of a particular group.  Is that possible with the existing implementation?

IE:
authenticator.name=ldap
ldap.url=ldaps://ldapvip.domain.com:636
ldap.bind-dn=CN=svc_user_ldap,OU=Service_Accounts,OU=Some_Accounts,DC=domain,DC=com
ldap.bind-password=some_password
ldap.user-bind-pattern=${USER}@domain.com
ldap.user-base-dn=OU=Some_Accounts,DC=domain,DC=com
ldap.group-auth-pattern=(&(objectClass=person)(sAMAccountName=${USER})(memberof=CN=pii_access_group,OU=Unix Groups,OU=Groups,OU=Some_Accounts,DC=domain,DC=com))

**Comment by Haarolean** [comment, marked as answer]
Hey, thanks for the feedback! Glad to hear you like the app.

Regarding the LDAP, may I ask, where have you got these properties? We got a few more which, as far as I understand, may cover all your needs, they're all located [here](https://github.com/provectus/kafka-ui/blob/master/documentation/compose/auth-ldap.yaml#L35).

**Reply by rja1** [reply]
Thanks @Haarolean, those parameters were just was I was looking for!  I'll give them a try..

The parameters I listed were  just an example of a typical configuration, from an apache trino cluster (I should have mentioned that).

Thanks again!
